src/cli/context.js

```diff
@@ -0,0 +1,457 @@
+"use strict";
+
+const path = require("path");
+const minimist = require("minimist");
+const fs = require("fs");
+const globby = require("globby");
+const chalk = require("chalk");
+const readline = require("readline");
+
+const util = require("./util");
+const normalizer = require("../main/options-normalizer");
+const prettier = require("../../index");
+const cleanAST = require("../common/clean-ast").cleanAST;
+const errors = require("../common/errors");
+const resolver = require("../config/resolve-config");
+const constant = require("./constant");
+const optionsModule = require("../main/options");
+const optionsNormalizer = require("../main/options-normalizer");
+const thirdParty = require("../common/third-party");
+const getSupportInfo = require("../common/support").getSupportInfo;
+const commonUtil = require("../common/util");
+
+const OPTION_USAGE_THRESHOLD = 25;
+const CHOICE_USAGE_MARGIN = 3;
+const CHOICE_USAGE_INDENTATION = 2;
+
+class Context {
```

Comment:
I'm not a fan of a context god class. Would be much nicer if it was a simple `cliContext` object that was passed to all functions.